cli/internal/helm/overrides.go

```diff
@@ -0,0 +1,156 @@
+		"measurementSalt":    base64.StdEncoding.EncodeToString(salt),
+		"attestationVariant": cfg.GetAttestationConfig().GetVariant().String(),
+		"attestationConfig":  string(attestationConfigJSON),
+	}
+	extraVals["verification-service"] = map[string]any{
+		"attestationVariant": cfg.GetAttestationConfig().GetVariant().String(),
+		"loadBalancerIP":     output.IP,
+	}
+	extraVals["konnectivity"] = map[string]any{
+		"loadBalancerIP": output.IP,
+	}
+
+	extraVals["key-service"] = map[string]any{
+		"masterSecret": base64.StdEncoding.EncodeToString(masterSecret),
+		"salt":         base64.StdEncoding.EncodeToString(salt),
+	}
+	switch cfg.GetProvider() {
+	case cloudprovider.OpenStack:
+		extraVals["openstack"] = map[string]any{
+			"deployYawolLoadBalancer": cfg.DeployYawolLoadBalancer(),
+		}
+		if cfg.DeployYawolLoadBalancer() {
+			extraVals["yawol-controller"] = map[string]any{
+				"yawolOSSecretName": "yawolkey",
+				// has to be larger than ~30s to account for slow OpenStack API calls.
+				"openstackTimeout": "1m",
+				"yawolFloatingID":  cfg.Provider.OpenStack.FloatingIPPoolID,
+				"yawolFlavorID":    cfg.Provider.OpenStack.YawolFlavorID,
+				"yawolImageID":     cfg.Provider.OpenStack.YawolImageID,
+			}
+		}
+	case cloudprovider.GCP:
+		serviceAccountKey, err := gcpshared.ServiceAccountKeyFromURI(serviceAccURI)
+		if err != nil {
+			return nil, fmt.Errorf("getting service account key: %w", err)
+		}
+		rawKey, err := json.Marshal(serviceAccountKey)
+		if err != nil {
+			return nil, fmt.Errorf("marshaling service account key: %w", err)
+		}
+		if output.GCP == nil {
+			return nil, fmt.Errorf("no GCP output from Terraform")
+		}
+		extraVals["ccm"] = map[string]any{
+			"GCP": map[string]any{
+				"projectID":         output.GCP.ProjectID,
+				"uid":               uid,
+				"secretData":        string(rawKey),
+				"subnetworkPodCIDR": output.GCP.IPCidrPod,
+			},
+		}
+	case cloudprovider.Azure:
+		if output.Azure == nil {
+			return nil, fmt.Errorf("no Azure output from Terraform")
+		}
+		ccmConfig, err := getCCMConfig(*output.Azure, serviceAccURI)
+		if err != nil {
+			return nil, fmt.Errorf("getting Azure CCM config: %w", err)
+		}
+		extraVals["ccm"] = map[string]any{
+			"Azure": map[string]any{
+				"azureConfig": string(ccmConfig),
+			},
+		}
+	}
+
+	return extraVals, nil
+}
+
+// extraOperatorValues returns the values for the constellation-operator chart.
+func extraOperatorValues(uid string) map[string]any {
+	return map[string]any{
+		"constellation-operator": map[string]any{
+			"constellationUID": uid,
+		},
+	}
+}
+
+// extraCSIValues returns the values for the csi chart.
+func extraCSIValues(provider cloudprovider.Provider, serviceAccURI string) (map[string]any, error) {
+	var csiVals map[string]any
+	if provider == cloudprovider.OpenStack {
+		creds, err := openstack.AccountKeyFromURI(serviceAccURI)
+		if err != nil {
+			return nil, err
+		}
+		cinderIni := creds.CloudINI().CinderCSIConfiguration()
```

Comment:
Same as with `gcpshared`.

Reply:
good suggestion. this is not marshalling though. Any idea for a general name of serviceAccURI then? marshalledServiceAccURI seems from wrong then